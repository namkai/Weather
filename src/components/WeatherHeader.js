import PropTypes from 'prop-types';
import React from 'react';

const WeatherHeader = ({ city, description }) => (
  <div className="dashhead">
    <div className="dashhead-titles">
      <h2 className="dashhead-title">{city}</h2>
      <h6 className="dashhead-subtitle" style={{ color: '#00BCD4' }}>Current conditions: {description[0].description}</h6>
    </div>

    <div className="btn-toolbar dashhead-toolbar">
      <div className="btn-toolbar-item input-with-icon">
        <input readOnly type="text" value={`    ${new Date().toDateString()}`} className="form-control" data-provide="datepicker" />
        <span className="icon icon-calendar" />
      </div>
    </div>
  </div>
);

WeatherHeader.defaultProps = {
	city: 'Weather',
	description: 'Get the weather in any city',
};

WeatherHeader.propTypes = {
	city: PropTypes.string,
	description: PropTypes.array,
};

export default WeatherHeader;
