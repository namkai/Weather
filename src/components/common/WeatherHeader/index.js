import WeatherHeader from './view';

export default WeatherHeader;
