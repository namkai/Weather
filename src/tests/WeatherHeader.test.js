import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import WeatherHeader from '../components/WeatherHeader';

it('renders correctly', () => {
	const cities = shallow(<WeatherHeader />);
	expect(toJson(cities)).toMatchSnapshot();
});