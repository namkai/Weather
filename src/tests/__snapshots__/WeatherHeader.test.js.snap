// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dashhead"
>
  <div
    className="dashhead-titles"
  >
    <h2
      className="dashhead-title"
    >
      Weather
    </h2>
    <h6
      className="dashhead-subtitle"
    />
  </div>
  <div
    className="btn-toolbar dashhead-toolbar"
  >
    <div
      className="btn-toolbar-item input-with-icon"
    >
      <input
        className="form-control"
        data-provide="datepicker"
        readOnly={true}
        type="text"
        value="    Sat Jun 03 2017"
      />
      <span
        className="icon icon-calendar"
      />
    </div>
  </div>
</div>
`;
